compare.py

```diff
@@ -0,0 +1,413 @@
+    current_length = len(boxes)
+    iterations = []
+    recurse = True
+
+    while current_length != previous_length and recurse:
+        for boxa, boxb in itertools.combinations(boxes, 2):
+            if intersecting_squares((boxa[0] - blur, boxa[1] - blur), (boxa[2] + blur, boxa[3] + blur),
+                                    (boxb[0] - blur, boxb[1] - blur), (boxb[2] + blur, boxb[3] + blur)):
+                logging.debug("boxa:{}".format(boxa))
+                logging.debug("boxb:{}".format(boxb))
+
+                box_ab = union(boxa, boxb)
+                if boxa in boxes:
+                    logging.debug("removing boxa: {}".format(boxa))
+                    boxes.remove(boxa)
+                    ignore.append(boxa)
+
+                if boxb in boxes:
+                    logging.debug("removing boxb: {}".format(boxb))
+                    boxes.remove(boxb)
+                    ignore.append(boxb)
+
+                if box_ab not in boxes:
+                    logging.debug(("boxes: {}".format(boxes)))
+                    boxes.append(box_ab)
+                    logging.debug(("appending boxab: {}".format(box_ab)))
+
+        iterations.append(len(boxes))
+
+        if len(iterations) > 2:
+            if iterations[-1] == iterations[-2] == iterations[-3]:
+                recurse = False
+
+        logging.debug("iterations={}".format(iterations))
+
+    return np.array(boxes).astype('int')
+
+
+def between(lower, upper, datum):
+    return upper >= datum >= lower
+
+
+def intersecting(r1_bottom_left, r1_top_right, r2_bottom_left, r2_top_right):
+    x_lower = r1_bottom_left[0]
+    x_upper = r1_top_right[0]
+    y_lower = r1_bottom_left[1]
+    y_upper = r1_top_right[1]
+
+    return ((between(lower=x_lower, upper=x_upper, datum=r2_bottom_left[0])
+            or between(lower=x_lower, upper=x_upper, datum=r2_top_right[0]))
+            and between(lower=y_lower, upper=y_upper, datum=r2_bottom_left[1])
+            or between(lower=y_lower, upper=y_upper, datum=r2_top_right[1]))
+
+
+def intersecting_squares(r1_bottom_left, r1_top_right, r2_bottom_left, r2_top_right):
+
+    return (intersecting(r1_bottom_left, r1_top_right, r2_bottom_left, r2_top_right)
+            or intersecting(r2_bottom_left, r2_top_right, r1_bottom_left, r1_top_right))
+
+
+def create_jira_ticket(summary, attachments, description, url, issuetype='Task', username='xxxxxxxxxxxxxxxxxx',
+                       api_key='xxxxxxxxxxxxxxxxxxxxxxxxxx', server='https://xxxxxxx.atlassian.net'):
+    new_issue = {}
+    jira_options = {'server': server}
+    jira_client = JIRA(options=jira_options, basic_auth=(username, api_key))
+
+    web_monitor_project = {'key': 'WM', 'name': 'Web Monitor', 'id': '12615'}
+
+    issue_dict = {
+        'project': {'id': web_monitor_project['id']},
+        'summary': summary,
+        'description': description,
+        'issuetype': {'name': issuetype},
+        'customfield_11833': url
+    }
+
+    try:
```

Comment:
multiple try except blocks not required for create and attachment , error description will anyways tell if its create error or attach error